### PR TITLE
Add X500 name constraints for non-organisation attributes

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

File: tools/demobench/src/main/kotlin/net/corda/demobench/views/NodeTabView.kt [hunk 1]
```diff
@@ -211,7 +211,7 @@ class NodeTabView : Fragment() {
                 error("Node with this name already exists")
             } else {
                 try {
-                    LegalNameValidator.validateLegalName(LegalNameValidator.normaliseLegalName(it))
+                    LegalNameValidator.validateOrganisation(LegalNameValidator.normaliseLegalName(it))
```
Comment (patrickkuo) [inline tools/demobench/src/main/kotlin/net/corda/demobench/views/NodeTabView.kt]: let's change "normaliseLegalName" as well? Maybe just `normalise`

**Comment by rnicoll** [comment]
Ready for re-review @patrickkuo

File: core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt [hunk 2]
```diff
@@ -45,7 +45,7 @@ data class CordaX500Name(val commonName: String?,
 
     init {
         // Legal name checks.
-        LegalNameValidator.validateLegalName(organisation)
+        LegalNameValidator.validateOrganization(organisation)
```
Comment (patrickkuo) [inline core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: LegalNameValidator.validateOrganization(**_organisation_**)

Comment (rnicoll) [reply core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: Yeah... that's my screwup... is it worth adding a deprecated `organisation` value and changing to US spelling (which is how it's spelt in the standard)?

Comment (rnicoll) [reply core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: Fixing the name in `CordaX500Name` is... lets say rather invasive. Going to skip for now and tackle later on.

**Pull request merged**